metricflow/specs/where_filter_entity.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+
+from typing import List, Sequence
+
+from dbt_semantic_interfaces.call_parameter_sets import (
+    EntityCallParameterSet,
+    FilterCallParameterSets,
+)
+from dbt_semantic_interfaces.naming.dundered import DunderedNameFormatter
+from dbt_semantic_interfaces.protocols.protocol_hint import ProtocolHint
+from dbt_semantic_interfaces.references import EntityReference
+from typing_extensions import override
+
+from metricflow.specs.column_assoc import ColumnAssociationResolver
+from metricflow.specs.query_interface import QueryInterfaceDimension, QueryInterfaceEntityFactory
+from metricflow.specs.specs import EntitySpec
+
+
+class WhereFilterEntity(ProtocolHint[QueryInterfaceDimension]):
+    """An entity that is passed in through the where filter parameter."""
+
+    @override
+    def _implements_protocol(self) -> QueryInterfaceDimension:
+        return self
+
+    def __init__(self, column_name: str):  # noqa
+        self.column_name = column_name
+
+    def grain(self, _grain: str) -> WhereFilterEntity:
+        """The time granularity."""
+        raise NotImplementedError
+
+    def alias(self, _alias: str) -> WhereFilterEntity:
+        """Renaming the column."""
+        raise NotImplementedError
+
+    def __str__(self) -> str:
+        """Returns the column name.
+
+        Important in the Jinja sandbox.
+        """
```

Comment:
Is this part of the contract for all QueryInterfaceDimension classes? If so, we might consider marking `__str__` as abstract in the protocol, although it's not clear to me if that'll actually do anything from a typechecking perspective. But if it does do something in the typechecker it'd be nicely self-documenting.

If this is specific to this implementation - meaning we normally don't care about being able to use implementing classes in a jinja sandbox - then the current layout is what we want.

Reply:
Actually, I was hoping that I could remove the `__str__` implementation in the Jinja-MFS interface after this PR is merged in.